datafusion/physical-plan/src/repartition/mod.rs

```diff
@@ -1427,9 +1430,12 @@ struct PerPartitionStream {
 
     /// Execution metrics
     baseline_metrics: BaselineMetrics,
+
+    batch_coalescer: Option<LimitedBatchCoalescer>,
```

Comment:
I recommend adding some context here about when this could be null -- something like

```suggestion
    /// None for sort preserving variant (merge sort already does coalescing)
    batch_coalescer: Option<LimitedBatchCoalescer>,
```